### PR TITLE
SM-6251: Add response enums that include upcoming_enum

### PR DESCRIPTION
**Pull request by caoimheb**
## Description

Please explain briefly the changes you made here.

## Coding Standards

Style guide: https://github.com/departmentfortransport/street-manager/wiki/Street-Manager-Style-Guide
Coding best practices: https://github.com/departmentfortransport/street-manager/wiki/Street-Manager-Coding-Best-Practices

## Checklist

- [ ] Branch named {feature|hotfix|task}/{SM-.*}
- [ ] If your pull request depends on any other, please link them
- [ ] Quality Street approval
- [ ] Street Smart approval
- [ ] API definitions updated
- [ ] Commit messages are meaningful
- [ ] Int UI tests passing https://circleci.com/gh/departmentfortransport/street-manager-int/tree/master
- [ ] Add `DO NOT MERGE` if you want to postpone merge

### CODE REVIEW

File: dist/interfaces/referenceTypes.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export declare enum AdditionalSpecialDesignationCode {
+export declare enum AdditionalSpecialDesignationCodeResponse {
```
Comment (caoimheb) [inline dist/interfaces/referenceTypes.d.ts:1]: These enums are only used in Response objects so don't need a duplicate for requests. Renamed for consistency

File: dist/interfaces/referenceTypes.d.ts [hunk 2]
```diff
@@ -22,9 +22,10 @@ export declare enum AdditionalSpecialDesignationCode {
     local_considerations = 27,
     winter_maintenance_routes = 28,
     hgv_approved_routes = 29,
-    emergency_services_routes = 30
+    emergency_services_routes = 30,
+    upcoming_enum = 999
```
Comment (caoimheb) [inline dist/interfaces/referenceTypes.d.ts:26]: These enums defined by sequential codes in the NSG data format. Making the assumption that if any more are added they will continue to be sequential.

File: src/interfaces/referenceTypes.ts [hunk 3]
```diff
@@ -1,4 +1,6 @@
-export enum AdditionalSpecialDesignationCode {
+// TODO Only used in responses - rename?
```
Comment (cpaul92) [inline src/interfaces/referenceTypes.ts:1]: Remove

File: package-lock.json [hunk 4]
```diff
@@ -2,7 +2,6 @@
   "name": "street-manager-street-lookup-client",
   "version": "1.0.0",
   "lockfileVersion": 1,
-  "requires": true,
```
Comment (cpaul92) [inline package-lock.json:5]: This change intentional?

File: src/interfaces/referenceTypes.ts (hunk 3 above)
Comment (caoimheb) [reply src/interfaces/referenceTypes.ts:1]: 🙈

**Review by Kellym-Kainos — APPROVED** [review]
Approved pending catherines comments

**Review by cpaul92 — APPROVED** [review]

**Pull request merged**